src/providers/hana/qgshanaprovider.cpp

```diff
@@ -1363,12 +1381,24 @@ QgsRectangle QgsHanaProvider::estimateExtent() const
         ret.setXMaximum( rsExtent->getValue( 3 ).toDouble() );
         ret.setYMaximum( rsExtent->getValue( 4 ).toDouble() );
       }
+      else if ( useEstimatedMetadata )
+      {
+        rsExtent->close();
+
+        // In this case it is very likely that the fast extent is not yet available, try again without using cached data
+        return estimateExtent( false );
+      }
     }
     rsExtent->close();
     return ret;
   }
   catch ( const QgsHanaException &ex )
   {
+    if ( useEstimatedMetadata ) {
+      // Try again without fast extent estimation
+      return estimateExtent( false );
```

Comment:
This recursion is dangerous. Why do you expect an exception for the fast extent estimation case?

Reply:
My mistake, I was not sure in which cases the exception is thrown. Removed.